doc/dev/release_checklist.md

```diff
@@ -0,0 +1,134 @@
+    - `REPL` depends on `Mathlib` (this dependency is only for testing).
+- Finally, make an announcement!
+  This should go in https://leanprover.zulipchat.com/#narrow/stream/113486-announce, with topic `v4.6.0`.
+  Please see previous announcements for suggested language.
+  You will want a few bullet points for main topics from the release notes.
+  If you are making a blog post, that should be coordinated with the announcement on Zulip and linked from there.
+  Please also make sure that whoever is handling social media knows the release is out.
+
+## Time estimates:
+- Initial checks and push the tag: 30 minutes.
+- Note that if `RELEASES.md` has discrepancies this could take longer!
+- Waiting for the release: 60 minutes.
+- Fixing release notes: 15 minutes.
+- Bumping toolchains in downstream repositories, up to craeting the Mathlib PR: 30 minutes.
+- Waiting for Mathlib CI and bors: 120 minutes.
+- Finalising Mathlib tags and stable branch, and updating REPL: 15 minutes.
+- Posting announcement and/or blog post: 20 minutes.
+
+# Creating a release candidate.
+
+This checklist walks you through creating the first release candidate for a version of Lean.
+
+We'll use `v4.7.0-rc1` as the intended release version in this example.
+
+- Decide which nightly release you want to turn into a release candidate.
+  We will use `nightly-2024-02-29` in this example.
+- It is essential that Std and Mathlib already have reviewed branches compatible with this nightly.
+  - Check that both Std and Mathlib's `bump/v4.7.0` branch contain `nightly-2024-02-29`
+    in their `lean-toolchain`.
+  - The steps required to reach that state are beyond the scope of this checklist!
+- Create the release branch from this nightly tag:
+    ```
+    git remote add nightly https://github.com/leanprover/lean4-nightly.git
+    git fetch nightly tag nightly-2024-02-29
+    git checkout nightly-2024-02-29
+    git checkout -b releases/v4.7.0
+    ```
+- In `RELEASES.md` remove `(development in progress)` from the `v4.7.0` section header.
+- Unfortunately, we are not yet consistent about updating `RELEASES.md` as part of each feature PR, so there may be manual work at this stage updating it.
+  - Do not wait on `RELEASES.md` being perfect before creating the `release/v4.7.0` branch. It is essential to fix the nightly which will become the release candidate as early as possible, to avoid confusion.
+  - You may like to solicit updates to `RELEASES.md` from other developers. Ideally these will land on `master` before the nightly.
+  - I will usually go through the list of all PRs merged since the previous release branch `releases/v4.6.0` (TODO: suggest a URL).
+  Many PRs do not warrant mention in `RELEASES.md`. For many medium scale PRs, it suffices to just copy and paste the title, along with a link. (See existing format in `RELEASES.md`.) If you find something major, it may require correspondence with the author of that PR to prepare a good paragraph.
```

Comment:
```suggestion
  Many PRs do not warrant mention in `RELEASES.md`. For many medium scale PRs, it suffices to just copy and paste the title, along with a link. (See existing format in `RELEASES.md`.) If you find something major that has not been mentioned, it may require correspondence with the author of that PR to prepare a good paragraph.
```